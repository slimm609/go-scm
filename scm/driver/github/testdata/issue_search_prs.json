{
  "total_count": 4,
  "incomplete_results": false,
  "items": [
    {
      "url": "https://api.github.com/repos/jenkins-x/jenkins-x-boot-config/issues/66",
      "repository_url": "https://api.github.com/repos/jenkins-x/jenkins-x-boot-config",
      "labels_url": "https://api.github.com/repos/jenkins-x/jenkins-x-boot-config/issues/66/labels{/name}",
      "comments_url": "https://api.github.com/repos/jenkins-x/jenkins-x-boot-config/issues/66/comments",
      "events_url": "https://api.github.com/repos/jenkins-x/jenkins-x-boot-config/issues/66/events",
      "html_url": "https://github.com/jenkins-x/jenkins-x-boot-config/pull/66",
      "id": 492638480,
      "node_id": "MDExOlB1bGxSZXF1ZXN0MzE2NzcxODM2",
      "number": 66,
      "title": "fix: add velero support",
      "user": {
        "login": "jstrachan",
        "id": 30140,
        "node_id": "MDQ6VXNlcjMwMTQw",
        "avatar_url": "https://avatars1.githubusercontent.com/u/30140?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/jstrachan",
        "html_url": "https://github.com/jstrachan",
        "followers_url": "https://api.github.com/users/jstrachan/followers",
        "following_url": "https://api.github.com/users/jstrachan/following{/other_user}",
        "gists_url": "https://api.github.com/users/jstrachan/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/jstrachan/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/jstrachan/subscriptions",
        "organizations_url": "https://api.github.com/users/jstrachan/orgs",
        "repos_url": "https://api.github.com/users/jstrachan/repos",
        "events_url": "https://api.github.com/users/jstrachan/events{/privacy}",
        "received_events_url": "https://api.github.com/users/jstrachan/received_events",
        "type": "User",
        "site_admin": false
      },
      "labels": [
        {
          "id": 1515633386,
          "node_id": "MDU6TGFiZWwxNTE1NjMzMzg2",
          "url": "https://api.github.com/repos/jenkins-x/jenkins-x-boot-config/labels/size/M",
          "name": "size/M",
          "color": "ededed",
          "default": false
        }
      ],
      "state": "open",
      "locked": false,
      "assignee": null,
      "assignees": [],
      "milestone": null,
      "comments": 7,
      "created_at": "2019-09-12T07:39:33Z",
      "updated_at": "2019-09-12T13:37:20Z",
      "closed_at": null,
      "author_association": "MEMBER",
      "pull_request": {
        "url": "https://api.github.com/repos/jenkins-x/jenkins-x-boot-config/pulls/66",
        "html_url": "https://github.com/jenkins-x/jenkins-x-boot-config/pull/66",
        "diff_url": "https://github.com/jenkins-x/jenkins-x-boot-config/pull/66.diff",
        "patch_url": "https://github.com/jenkins-x/jenkins-x-boot-config/pull/66.patch"
      },
      "body": "",
      "score": 1.0
    },
    {
      "url": "https://api.github.com/repos/jenkins-x/jx/issues/5452",
      "repository_url": "https://api.github.com/repos/jenkins-x/jx",
      "labels_url": "https://api.github.com/repos/jenkins-x/jx/issues/5452/labels{/name}",
      "comments_url": "https://api.github.com/repos/jenkins-x/jx/issues/5452/comments",
      "events_url": "https://api.github.com/repos/jenkins-x/jx/issues/5452/events",
      "html_url": "https://github.com/jenkins-x/jx/pull/5452",
      "id": 492709862,
      "node_id": "MDExOlB1bGxSZXF1ZXN0MzE2ODI5MzM3",
      "number": 5452,
      "title": "fix: avoid failing boot if folks want TLS and are not using GKE",
      "user": {
        "login": "jstrachan",
        "id": 30140,
        "node_id": "MDQ6VXNlcjMwMTQw",
        "avatar_url": "https://avatars1.githubusercontent.com/u/30140?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/jstrachan",
        "html_url": "https://github.com/jstrachan",
        "followers_url": "https://api.github.com/users/jstrachan/followers",
        "following_url": "https://api.github.com/users/jstrachan/following{/other_user}",
        "gists_url": "https://api.github.com/users/jstrachan/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/jstrachan/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/jstrachan/subscriptions",
        "organizations_url": "https://api.github.com/users/jstrachan/orgs",
        "repos_url": "https://api.github.com/users/jstrachan/repos",
        "events_url": "https://api.github.com/users/jstrachan/events{/privacy}",
        "received_events_url": "https://api.github.com/users/jstrachan/received_events",
        "type": "User",
        "site_admin": false
      },
      "labels": [
        {
          "id": 1012676885,
          "node_id": "MDU6TGFiZWwxMDEyNjc2ODg1",
          "url": "https://api.github.com/repos/jenkins-x/jx/labels/size/S",
          "name": "size/S",
          "color": "ededed",
          "default": false
        }
      ],
      "state": "open",
      "locked": false,
      "assignee": null,
      "assignees": [],
      "milestone": null,
      "comments": 5,
      "created_at": "2019-09-12T10:05:55Z",
      "updated_at": "2019-09-12T19:56:58Z",
      "closed_at": null,
      "author_association": "MEMBER",
      "pull_request": {
        "url": "https://api.github.com/repos/jenkins-x/jx/pulls/5452",
        "html_url": "https://github.com/jenkins-x/jx/pull/5452",
        "diff_url": "https://github.com/jenkins-x/jx/pull/5452.diff",
        "patch_url": "https://github.com/jenkins-x/jx/pull/5452.patch"
      },
      "body": "we log a warning that the combination has not been tested though\n\nfixes #5451\n\nSigned-off-by: James Strachan <james.strachan@gmail.com>",
      "score": 1.0
    },
    {
      "url": "https://api.github.com/repos/jenkins-x/jx/issues/5466",
      "repository_url": "https://api.github.com/repos/jenkins-x/jx",
      "labels_url": "https://api.github.com/repos/jenkins-x/jx/issues/5466/labels{/name}",
      "comments_url": "https://api.github.com/repos/jenkins-x/jx/issues/5466/comments",
      "events_url": "https://api.github.com/repos/jenkins-x/jx/issues/5466/events",
      "html_url": "https://github.com/jenkins-x/jx/pull/5466",
      "id": 492916054,
      "node_id": "MDExOlB1bGxSZXF1ZXN0MzE2OTk2MTA3",
      "number": 5466,
      "title": "fix: a `jx bdd` command to trigger the bdd tests",
      "user": {
        "login": "jstrachan",
        "id": 30140,
        "node_id": "MDQ6VXNlcjMwMTQw",
        "avatar_url": "https://avatars1.githubusercontent.com/u/30140?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/jstrachan",
        "html_url": "https://github.com/jstrachan",
        "followers_url": "https://api.github.com/users/jstrachan/followers",
        "following_url": "https://api.github.com/users/jstrachan/following{/other_user}",
        "gists_url": "https://api.github.com/users/jstrachan/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/jstrachan/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/jstrachan/subscriptions",
        "organizations_url": "https://api.github.com/users/jstrachan/orgs",
        "repos_url": "https://api.github.com/users/jstrachan/repos",
        "events_url": "https://api.github.com/users/jstrachan/events{/privacy}",
        "received_events_url": "https://api.github.com/users/jstrachan/received_events",
        "type": "User",
        "site_admin": false
      },
      "labels": [
        {
          "id": 1013488046,
          "node_id": "MDU6TGFiZWwxMDEzNDg4MDQ2",
          "url": "https://api.github.com/repos/jenkins-x/jx/labels/do-not-merge/hold",
          "name": "do-not-merge/hold",
          "color": "ededed",
          "default": false
        },
        {
          "id": 1016466107,
          "node_id": "MDU6TGFiZWwxMDE2NDY2MTA3",
          "url": "https://api.github.com/repos/jenkins-x/jx/labels/size/L",
          "name": "size/L",
          "color": "ededed",
          "default": false
        }
      ],
      "state": "open",
      "locked": false,
      "assignee": null,
      "assignees": [],
      "milestone": null,
      "comments": 11,
      "created_at": "2019-09-12T16:42:36Z",
      "updated_at": "2019-09-13T11:09:49Z",
      "closed_at": null,
      "author_association": "MEMBER",
      "pull_request": {
        "url": "https://api.github.com/repos/jenkins-x/jx/pulls/5466",
        "html_url": "https://github.com/jenkins-x/jx/pull/5466",
        "diff_url": "https://github.com/jenkins-x/jx/pull/5466.diff",
        "patch_url": "https://github.com/jenkins-x/jx/pull/5466.patch"
      },
      "body": "so folks can easily run the BDD tests inside any Jenkins X installation\r\n\r\nfixes https://github.com/jenkins-x/jx/issues/5454",
      "score": 1.0
    },
    {
      "url": "https://api.github.com/repos/jenkins-x/go-scm/issues/29",
      "repository_url": "https://api.github.com/repos/jenkins-x/go-scm",
      "labels_url": "https://api.github.com/repos/jenkins-x/go-scm/issues/29/labels{/name}",
      "comments_url": "https://api.github.com/repos/jenkins-x/go-scm/issues/29/comments",
      "events_url": "https://api.github.com/repos/jenkins-x/go-scm/issues/29/events",
      "html_url": "https://github.com/slimm609/go-scm/pull/29",
      "id": 493196740,
      "node_id": "MDExOlB1bGxSZXF1ZXN0MzE3MjIxMDEw",
      "number": 29,
      "title": "fix: add another example program",
      "user": {
        "login": "jstrachan",
        "id": 30140,
        "node_id": "MDQ6VXNlcjMwMTQw",
        "avatar_url": "https://avatars1.githubusercontent.com/u/30140?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/jstrachan",
        "html_url": "https://github.com/jstrachan",
        "followers_url": "https://api.github.com/users/jstrachan/followers",
        "following_url": "https://api.github.com/users/jstrachan/following{/other_user}",
        "gists_url": "https://api.github.com/users/jstrachan/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/jstrachan/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/jstrachan/subscriptions",
        "organizations_url": "https://api.github.com/users/jstrachan/orgs",
        "repos_url": "https://api.github.com/users/jstrachan/repos",
        "events_url": "https://api.github.com/users/jstrachan/events{/privacy}",
        "received_events_url": "https://api.github.com/users/jstrachan/received_events",
        "type": "User",
        "site_admin": false
      },
      "labels": [
        {
          "id": 1512534528,
          "node_id": "MDU6TGFiZWwxNTEyNTM0NTI4",
          "url": "https://api.github.com/repos/jenkins-x/go-scm/labels/size/M",
          "name": "size/M",
          "color": "ededed",
          "default": false
        }
      ],
      "state": "open",
      "locked": false,
      "assignee": null,
      "assignees": [],
      "milestone": null,
      "comments": 1,
      "created_at": "2019-09-13T08:21:21Z",
      "updated_at": "2019-09-13T08:21:37Z",
      "closed_at": null,
      "author_association": "MEMBER",
      "pull_request": {
        "url": "https://api.github.com/repos/jenkins-x/go-scm/pulls/29",
        "html_url": "https://github.com/slimm609/go-scm/pull/29",
        "diff_url": "https://github.com/slimm609/go-scm/pull/29.diff",
        "patch_url": "https://github.com/slimm609/go-scm/pull/29.patch"
      },
      "body": "add a little CLI tool to lookup a PR",
      "score": 1.0
    }
  ]
}
