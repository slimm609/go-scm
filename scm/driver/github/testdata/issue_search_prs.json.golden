[
  {
    "Number": 66,
    "Title": "fix: add velero support",
    "Body": "",
    "Link": "https://github.com/jenkins-x/jenkins-x-boot-config/pull/66",
    "State": "open",
    "Labels": [
      "size/M"
    ],
    "Closed": false,
    "Locked": false,
    "Author": {
      "Login": "jstrachan",
      "Name": "",
      "Email": "",
      "Avatar": "https://avatars1.githubusercontent.com/u/30140?v=4",
      "Link": "",
      "Created": "0001-01-01T00:00:00Z",
      "Updated": "0001-01-01T00:00:00Z"
    },
    "Assignees": null,
    "PullRequest": true,
    "Created": "2019-09-12T07:39:33Z",
    "Updated": "2019-09-12T13:37:20Z",
    "Repository": {
      "ID": "",
      "Namespace": "jenkins-x",
      "Name": "jenkins-x-boot-config",
      "FullName": "",
      "Perm": null,
      "Branch": "",
      "Private": false,
      "Clone": "",
      "CloneSSH": "",
      "Link": "",
      "Created": "0001-01-01T00:00:00Z",
      "Updated": "0001-01-01T00:00:00Z"
    }
  },
  {
    "Number": 5452,
    "Title": "fix: avoid failing boot if folks want TLS and are not using GKE",
    "Body": "we log a warning that the combination has not been tested though\n\nfixes #5451\n\nSigned-off-by: James Strachan \u003cjames.strachan@gmail.com\u003e",
    "Link": "https://github.com/jenkins-x/jx/pull/5452",
    "State": "open",
    "Labels": [
      "size/S"
    ],
    "Closed": false,
    "Locked": false,
    "Author": {
      "Login": "jstrachan",
      "Name": "",
      "Email": "",
      "Avatar": "https://avatars1.githubusercontent.com/u/30140?v=4",
      "Link": "",
      "Created": "0001-01-01T00:00:00Z",
      "Updated": "0001-01-01T00:00:00Z"
    },
    "Assignees": null,
    "PullRequest": true,
    "Created": "2019-09-12T10:05:55Z",
    "Updated": "2019-09-12T19:56:58Z",
    "Repository": {
      "ID": "",
      "Namespace": "jenkins-x",
      "Name": "jx",
      "FullName": "",
      "Perm": null,
      "Branch": "",
      "Private": false,
      "Clone": "",
      "CloneSSH": "",
      "Link": "",
      "Created": "0001-01-01T00:00:00Z",
      "Updated": "0001-01-01T00:00:00Z"
    }
  },
  {
    "Number": 5466,
    "Title": "fix: a `jx bdd` command to trigger the bdd tests",
    "Body": "so folks can easily run the BDD tests inside any Jenkins X installation\r\n\r\nfixes https://github.com/jenkins-x/jx/issues/5454",
    "Link": "https://github.com/jenkins-x/jx/pull/5466",
    "State": "open",
    "Labels": [
      "do-not-merge/hold",
      "size/L"
    ],
    "Closed": false,
    "Locked": false,
    "Author": {
      "Login": "jstrachan",
      "Name": "",
      "Email": "",
      "Avatar": "https://avatars1.githubusercontent.com/u/30140?v=4",
      "Link": "",
      "Created": "0001-01-01T00:00:00Z",
      "Updated": "0001-01-01T00:00:00Z"
    },
    "Assignees": null,
    "PullRequest": true,
    "Created": "2019-09-12T16:42:36Z",
    "Updated": "2019-09-13T11:09:49Z",
    "Repository": {
      "ID": "",
      "Namespace": "jenkins-x",
      "Name": "jx",
      "FullName": "",
      "Perm": null,
      "Branch": "",
      "Private": false,
      "Clone": "",
      "CloneSSH": "",
      "Link": "",
      "Created": "0001-01-01T00:00:00Z",
      "Updated": "0001-01-01T00:00:00Z"
    }
  },
  {
    "Number": 29,
    "Title": "fix: add another example program",
    "Body": "add a little CLI tool to lookup a PR",
    "Link": "https://github.com/slimm609/go-scm/pull/29",
    "State": "open",
    "Labels": [
      "size/M"
    ],
    "Closed": false,
    "Locked": false,
    "Author": {
      "Login": "jstrachan",
      "Name": "",
      "Email": "",
      "Avatar": "https://avatars1.githubusercontent.com/u/30140?v=4",
      "Link": "",
      "Created": "0001-01-01T00:00:00Z",
      "Updated": "0001-01-01T00:00:00Z"
    },
    "Assignees": null,
    "PullRequest": true,
    "Created": "2019-09-13T08:21:21Z",
    "Updated": "2019-09-13T08:21:37Z",
    "Repository": {
      "ID": "",
      "Namespace": "jenkins-x",
      "Name": "go-scm",
      "FullName": "",
      "Perm": null,
      "Branch": "",
      "Private": false,
      "Clone": "",
      "CloneSSH": "",
      "Link": "",
      "Created": "0001-01-01T00:00:00Z",
      "Updated": "0001-01-01T00:00:00Z"
    }
  }
]